services/circleci/circleci.service.js

```diff
@@ -1,21 +1,20 @@
 'use strict'
 
-const Joi = require('joi')
 const BaseJsonService = require('../base-json')
-
-const circleSchema = Joi.array()
-  .items(Joi.object({ status: Joi.string().required() }))
-  .min(1)
-  .max(1)
-  .required()
+const { InvalidResponse } = require('../errors')
+const {
+  circleSchema,
+  getLatestCompleteBuildOutcome,
+  summarizeBuildsForLatestCompleteWorkflow,
+} = require('./circleci.helpers.js')
 
 module.exports = class CircleCi extends BaseJsonService {
   async fetch({ token, vcsType, userRepo, branch }) {
     let url = `https://circleci.com/api/v1.1/project/${vcsType}/${userRepo}`
     if (branch != null) {
       url += `/tree/${branch}`
     }
-    const query = { filter: 'completed', limit: 1 }
+    const query = { limit: 50 }
```

Comment:
We can't filter on 'completed' any more here. We need to know about in-progress/queued builds so that if the first workflow we come across has some builds which are not complete we know to ignore that one and move on to the next workflow to try and find one where all the builds are complete. If we filter on complete here, sometimes we will summarise an incomplete workflow.

In terms of limit, I've picked 50 to balance 2 properties:
* We need a fairly large number of records so that on a project with a large build matrix where we can't take the first `workflow_id` (e.g: because some builds are still in progress) we've got enough records
* If we fetch _loads_ of records the badge takes a long time to render because we must wait a long time for the `fetch()` stage